### DISCUSSION TITLE
Simplify default env-template

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#22324] [Feature Requests]
### Describe the Request

The current [env-template](https://github.com/directus/directus/blob/main/api/src/cli/utils/create-env/env-stub.liquid) contains some of the most important configurations as well a link to all config options. 

When creating a new project it feels wrong that:
- some of the settings in the template are explicitly set to the default values
- while others simply aren't included (and therewith use the defaults too) 
- and others set different values than the default ones.

As directus global defaults (in the code) are set for a good security I'd suggest to clean this up to only include:

1. required configs / configs that should be in most of the cases (like mail)
2. Configs which most likely should be changed in dev (e.g the CORS, EXTENSIONS_AUTO_RELOAD, or all storage & mail related) with a comment

To keep it further cleaner I' recommend to not include that many examples for e.g db, storage and mail but just have the comment 

```.env
### File Storage
# A CSV of storage locations (eg: local,digitalocean,amazon) to use. You can use any names you'd like for these keys ["local"]
# TODO: we've set a local storage as a default. Make sure to mount this in docker or set an external storage. Check the docs for more informations
STORAGE_LOCATIONS="local"
STORAGE_LOCAL_DRIVER="local"
STORAGE_LOCAL_ROOT="./uploads"
```

Looking forward to your feedback!

### Maintainence Strategy

n/A

**Comment by br41nslug** [comment]
Moved this to a feature request as we'll have to descuss a bit more explicit detail here:
- Which environment variables are globally required
- Which environment variables are required for certain use-cases
- Which environment variables are not required but still useful

**Comment by Dominic-Marcelino** [comment]
@br41nslug Thanks for getting back. Following I'll try to answer the questions prom my pov.


## Main goals
**I'll share my thoughts by setting the my main goals first:**
1. Make the generated env file more readable & reduce clutter
2. More consistency: 
    - Don't include some global defaults while others are not included
    - Don't just override some global defaults without educating why. I'd expect the global defaults to be the most secure
3. Keep defaults up to date
    - As the current template includes some of the defaults, in case of breaking changes with updated defaults, I'd not be aligned with the global defaults anymore. Reading the changelog and the breaking changes I'd expect that these changes happen on update, in case I never changed the them manually
4. Educate people: 
    - The current config let users assume these are all available configs while there's a lot more
    - The current config could make the user assume these are important settings while others aren't needed

## Solutions
**To archive this, I'd suggest:**
1. Do not include any setting that falls back to the default value
2. Do not provide multiple examples
3. ...add "ToDo" comments instead on all configs that need to be set (or should be set in most cases)

## Example env-template
**That given the template could look like this:**

```.env
####################################################################################################
#
# These values set environment variables which modify core settings of Directus.
#
# For all available options see:
# https://docs.directus.io/self-hosted/config-options/
#
####################################################################################################
####################################################################################################

####################################################################################################
### DEV
# If you start developing you'd most likely want to have these in your dev instance.
# On production you most likely don't want to have them

# EXTENSIONS_AUTO_RELOAD=true
# REFRESH_TOKEN_COOKIE_SECURE=false
# SESSION_COOKIE_SECURE=false


####################################################################################################
### General
# TODO: set your host
HOST="0.0.0.0"
# TODO: set your port
PORT=8055
# Todo: set your public url
PUBLIC_URL="/"


####################################################################################################
### Database

# All DB_* environment variables are passed to the connection configuration of a Knex instance.
# Based on your project's needs, you can extend the DB_* environment variables with any config
# you need to pass to the database instance.

{{ database }}


####################################################################################################
### File Storage

# TODO: set your storage provider. If you use 'local' on docker make sure to mount the volume


####################################################################################################
### Security

{{ security }}


# TODO: set a custom cookie name for REFRESH_TOKEN_COOKIE_NAME and SESSION_COOKIE_NAME


####################################################################################################
### Email

# TODO: setup an email provider


```


## Discussions:
1. If it's recommended to increase the defaults for argon I'd suggest to increate them globally instead of in the env template
2. I'd suggest to enable `CORS` by default and add a disabled config to the `dev` section of the env file
3. I'd suggest to set cookies to secured-cookies by default and config to the `dev` section of the env file


Really interested in some insights / feedback from the core team ☺️

**Reply by Dominic-Marcelino** [reply]
Are there any thoughts / feedback on the proposal? ☺️

**Reply by br41nslug** [reply]
Looks like it would be a nice improvement, but I personally havent had much time to dive into this.
